src/VisualStudio/Core/Def/Remote/VisualStudioWorkspaceServiceHubConnector.cs

```diff
@@ -38,7 +40,7 @@ public void StartListening(Workspace workspace, object serviceOpt)
         }
 
         // only push solution snapshot from primary (VS) workspace:
-        _checksumUpdater = new SolutionChecksumUpdater(workspace, _listenerProvider, _disposalCancellationSource.Token);
+        _checksumUpdater = new SolutionChecksumUpdater(workspace, _listenerProvider, _threadingContext, _disposalCancellationSource.Token);
```

Comment:
If we have the threading context, we might be able to get rid of the disposal logic here and passing that ct along